#!/usr/bin/env python
PACKAGE = "tailored_mpc"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Ax_acc_max",   double_t,   0,  "Maximum longitudinal acceleration (when accelerating)",    8.0,    0.0,    20.0)
gen.add("Ax_dec_max",   double_t,   0,  "Maximum longitudinal acceleration (when braking)",         10.0,   0.0,    20.0)
gen.add("Ay_max",       double_t,   0,  "Maximum lateral acceleration",                             10.0,   0.0,    20.0)
gen.add("Vx_max",       double_t,   0,  "Maximum longitudinal velocity",                            20.0,   0.0,    30.0)
gen.add("Vx_final",     double_t,   0,  "Final velocity",                                           15.0,   0.0,    20.0)

 
exit(gen.generate(PACKAGE, "longitudinal_controller", "long"))